astroquery/heasarc/core.py

```diff
@@ -171,7 +171,7 @@ def query_region_async(self, position: Union[coordinates.SkyCoord, str],
         # Generate the request
         request_payload = self._args_to_payload(
             mission=mission,
-            entry="{},{}".format(c.ra.degree, c.dec.degree),
+            entry="{:10f},{:.10f}".format(c.ra.degree, c.dec.degree),
```

Comment:
I would suggest the following:
```suggestion
            entry=f"{c.ra.degree:xf},{c.deg.degree:.xf}",
```
where `x` is the value that ends up producing the current filenames. I don't know what that value is, but apparently it is not 10. The purpose of this change would be to ensure that identical user input gets converted to an identical request. Introducing a deliberate change would defeat that purpose.

Reply:
The problem that motivated this PR is that this "x" is not constant between software versions. So there is no way to set it uniquely.
Also I am not sure we are talking about the same thing. 
I want to make it so that for each single user input, the request sent to the service is the same for all software versions.
Is this what you mean?
Since the formatting apparently varies between versions, I need to introduce a deliberate, fixed, format.

Reply:
What I am suggesting is that the deliberate fixed format that you use should be the same that is being used automatically in the tests that currently pass on `main` because otherwise the user request will vary between `astrouery` versions.

Reply:
Alright, so we agree that a deliberate change, introducing constant format, is need, I was not sure. 
So the question is to what to set it.
I choose 1e-10 a bit arbitrarily, but considering the precision needed to request source tables. It's sub-micro-arcsec, it's enough. So it's scientifically motivated.
So, I can see how it could makes sense to preserve the behavior in the already merged version, even if it was not motivated by any requirement on precision.
I tried to deduce which it is, and in the test which broke here, it's `f"{c.ra.degree:.13f},{c.dec.degree:.16f}"`. Is this so useful to keep? I'd prefer to use motivated f"{c.ra.degree:.10f},{c.dec.degree:.10f}".
Furthermore, while setting `f"{c.ra.degree:.13f},{c.dec.degree:.16f}"` fixes the couple of broken tests, it breaks other tests!
For example, it represents `10, 10` as `10.0000000000000,10.0000000000000018`. Another kind of trouble while formatting close top precision.
This also can be avoided by using 10.
At a small cost of negligible change in the request between versions. And from now on, all requests should be consistent.

Reply:
If it is not possible to specify a precision that is always compatible with the current automatically used value then a one-time change in the requests must be accepted, and a sub-microarcsecond precision should be good enough.

Reply:
Ok!